plugins/statistics.koplugin/main.lua

```diff
@@ -2687,6 +2688,43 @@ function ReaderStatistics:getReadBookByDay(month)
     return per_day
 end
 
+function ReaderStatistics:getReadingDurationBySecond(date)
+    local sql_stmt = [[
+        SELECT
+            start_time - strftime('%s', ?, 'utc') as start,
+            start_time - strftime('%s', ?, 'utc') + duration as end,
+            id_book book_id,
+            book.title book_title
+        FROM   page_stat_data
```

Comment:
By using `page_stat_data`, I guess we avoid some computations done by the `page_stat` "view" (unless sqlite is clever enough to figure it's not needed to use it and it can just access the real table underneath).
May be some other SQL queries could benefit from also using `page_stat_data` ? (It's possible we/I made these SQL before the page_stat_data / page_stat view were introduced).

Reply:
>unless sqlite is clever enough to figure

I doubt that, but I'm not expert in the inner workings of sqlite...

>May be some other SQL queries could benefit from also using page_stat_data ?

I did a quick check and many of them use the `page` column, on which the page_stat view seem to be the better choice :)